runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/StreamingEngineClient.java

```diff
@@ -97,18 +104,19 @@ public final class StreamingEngineClient {
   private StreamingEngineClient(
       JobHeader jobHeader,
       GetWorkBudget totalGetWorkBudget,
-      AtomicReference<StreamingEngineConnectionState> connections,
       GrpcWindmillStreamFactory streamFactory,
-      WorkItemProcessor workItemProcessor,
+      WorkItemScheduler workItemScheduler,
       ChannelCachingStubFactory channelCachingStubFactory,
       GetWorkBudgetDistributor getWorkBudgetDistributor,
       GrpcDispatcherClient dispatcherClient,
-      long clientId) {
+      long clientId,
+      Function<WindmillStream.CommitWorkStream, WorkCommitter> workCommitterFactory,
+      Consumer<List<Windmill.ComputationHeartbeatResponse>> heartbeatResponseProcessor) {
     this.jobHeader = jobHeader;
     this.started = new AtomicBoolean();
```

Comment:
I think started could just be a boolean now that it is synchronized blocks.

Remove where it is set to true other than in start method (startMetadataStream).
Also consider just inlining startMetadataStream

Reply:
done

Reply:
finish method (like stop in StreamingDataflowWorker) is only there for testing so i added annotation there as well